workflows/prognostic_c48_run/prepare_config.py

```diff
@@ -101,80 +100,55 @@ def user_config_from_dict_and_args(config_dict: dict, args) -> UserConfig:
-    else:
-        nudging = None
 
-    scikit_learn = MachineLearningConfig(
-        model=list(args.model_url or []), diagnostic_ml=args.diagnostic_ml
-    )
+    user_config = dacite.from_dict(UserConfig, config_dict)
 
-    if "diagnostics" in config_dict:
-        diagnostics = [
-            dacite.from_dict(DiagnosticFileConfig, diag)
-            for diag in config_dict["diagnostics"]
-        ]
+    # insert command line option overrides
+    if user_config.scikit_learn is None:
+        if args.model_url:
+            user_config.scikit_learn = MachineLearningConfig(
+                model=list(args.model_url), diagnostic_ml=args.diagnostic_ml
+            )
     else:
-        diagnostics = _default_diagnostics(
-            nudging, scikit_learn, nudge_to_observations, args.output_frequency,
+        if args.model_url:
+            user_config.scikit_learn.model = list(args.model_url)
+        if args.diagnostic_ml:
+            user_config.scikit_learn.diagnostic_ml = args.diagnostic_ml
+
+    # insert custom default diagnostics
+    if len(user_config.diagnostics) == 0:
+        user_config.diagnostics = _default_diagnostics(
+            user_config.nudging,
+            user_config.scikit_learn,
+            nudge_to_observations,
+            args.output_frequency,
+        )
+    if len(user_config.fortran_diagnostics) == 0:
+        user_config.fortran_diagnostics = _default_fortran_diagnostics(
+            nudge_to_observations
         )
 
-    if "fortran_diagnostics" in config_dict:
-        fortran_diagnostics = [
-            dacite.from_dict(FortranFileConfig, diag)
-            for diag in config_dict["fortran_diagnostics"]
-        ]
-    else:
-        fortran_diagnostics = _default_fortran_diagnostics(nudge_to_observations)
-
-    default = UserConfig(diagnostics=[], fortran_diagnostics=[])
-
-    if nudging and len(scikit_learn.model):
+    if user_config.nudging and user_config.scikit_learn:
         raise NotImplementedError(
             "Nudging and machine learning cannot currently be run at the same time."
         )
 
-    return UserConfig(
-        prephysics=prephysics,
-        nudging=nudging,
-        diagnostics=diagnostics,
-        fortran_diagnostics=fortran_diagnostics,
-        scikit_learn=scikit_learn,
-        step_storage_variables=config_dict.get(
-            "step_storage_variables", default.step_storage_variables
-        ),
-        step_tendency_variables=config_dict.get(
-            "step_tendency_variables", default.step_tendency_variables
-        ),
-    )
+    return user_config
 
 
 def _default_diagnostics(
     nudging: Optional[NudgingConfig],
-    scikit_learn: MachineLearningConfig,
+    scikit_learn: Optional[MachineLearningConfig],
     nudge_to_obs: bool,
     frequency_minutes: int,
 ) -> List[DiagnosticFileConfig]:
     diagnostic_files: List[DiagnosticFileConfig] = []
 
-    if scikit_learn.model:
+    if scikit_learn:
```

Comment:
I always get confused about the true value of complex data structures. Can you use this instead?
```suggestion
    if scikit_learn is not None:
```